### PR TITLE
(feat): Add version numbers for apps and display them in the implementer tools

### PR DESCRIPTION
**Pull request by ibacher**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

This adds a constant, `__VERSION__` to our webpack build configuration which can be added to any module to get its current version. This also adds the convention that the `index.ts` of an app exports a `version` field to identify the version of the frontend module. Finally, I've added a tab to the implementer tools to give you an overview of the versions of modules running.

This requires modifications to existing apps that's something like this:

```ts
declare var __VERSION__: string;  // keep TS happy
const version = __VERSION__;  // __VERSION__ is replaced by Webpack with the version from package.json

// ...

export { /* existing stuff */, version };  // add the version to the exports
```

As shown, for modules that do not declare a version the string "unknown" is shown.

## Screenshots

Screenshot of implementer tools showing the version number view

### CODE REVIEW

File: packages/apps/esm-implementer-tools-app/src/frontend-modules/useFrontendModules.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        name: module[0],
+      }));
+  }
+
+  return cachedFrontendModules;
+}
+
+export function useFrontendModules() {
+  return getModules();
+}
```
Comment (ZacButko) [inline packages/apps/esm-implementer-tools-app/src/frontend-modules/useFrontendModules.ts:21]: Why have a separate function to repeat functionality of `getModules()` ?

**Review by ZacButko — APPROVED** [review]
This is amazing. Thank you so much @ibacher !

File: packages/apps/esm-implementer-tools-app/src/frontend-modules/useFrontendModules.ts (hunk 1 above)
Comment (ibacher) [reply packages/apps/esm-implementer-tools-app/src/frontend-modules/useFrontendModules.ts:21]: Oh... yeah that doesn't make much sense does it?

File: packages/apps/esm-devtools-app/src/index.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { getAsyncLifecycle } from "@openmrs/esm-framework";
 
+declare var __VERSION__: string;
+const version = __VERSION__;
```
[the remark this answers is not in the document]
Comment (ibacher) [reply packages/apps/esm-devtools-app/src/index.ts:4]: My original plan was to keep them in esm-template-app, but ok

**Pull request merged**